multimodal/src/autogluon/multimodal/matcher.py

```diff
@@ -102,13 +105,15 @@ class MultiModalMatcher:
 
     def __init__(
         self,
-        query: Union[str, List[str]],
-        response: Union[str, List[str]],
+        query: Optional[Union[str, List[str]]] = None,
+        response: Optional[Union[str, List[str]]] = None,
         negative: Optional[Union[str, List[str]]] = None,
         label: Optional[str] = None,
         match_label: Optional[Union[int, str]] = None,
         problem_type: Optional[str] = None,
+        pipeline: Optional[str] = None,
```

Comment:
we may use "search" as the pipeline for  ranking tasks, I guess "search" is more intuitive for general users.

Reply:
Users may use the `text_similarity` pipeline to do either text matching or text search.

Reply:
It feels a bit weird but we can revise later.